src/rrdp.rs

```diff
@@ -509,20 +459,19 @@ impl RrdpState {
         self.deprecated_files.push(DeprecatedFile::new(path));
     }
 
-    /// Writes any deltas for which no current file is found. I.e. it is assumed
-    /// that a file which is present was not tampered with since writing it.
-    fn write_missing_deltas(&self) -> Result<()> {
-        for delta in &self.deltas {
-            let path = Self::path_delta(&self.rrdp_dir, self.session_id(), delta.serial);
+    /// Writes new deltas. I.e. deltas for which we have XML in memory.
+    fn write_new_deltas(&mut self) -> Result<()> {
+        let session_id = self.session_id();
+        for delta in self.deltas.iter_mut() {
+            let path = Self::path_delta(&self.rrdp_dir, session_id, delta.serial);
             if path.exists() {
                 debug!("Skip writing delta file to {:?}", path)
             } else {
                 info!("Writing delta file to {:?}", path);
                 let xml = delta
-                    .xml()
-                    .ok_or_else(|| anyhow!("Delta XML not recovered on startup"))?;
-                file_ops::write_buf(&path, xml)
-                    .with_context(|| format!("Could not write delta XML to: {:?}", path))?;
+                    .take_xml()
+                    .ok_or_else(|| anyhow!("Delta XML no longer in memory, do not delete files! Restart from clean state!"))?;
+                file_ops::write_buf(&path, &xml).with_context(|| "Could not write delta XML")?;
```

Comment:
Did you mean to lose the path context in this error message?

Reply:
no, fixed